src/Modules/UngradedModules.jl

```diff
@@ -0,0 +1,3808 @@
+  end
+
+  R = base_ring(M)
+  #remove columns
+
+  M_generators = matrix(M.sub)
+  M_relations = isdefined(M, :quo) ? matrix(M.quo) : zero_matrix(R, 1,ncols(M_generators))
+
+  to_delete::Vector{Int} = []
+  for i=1:size(M_relations)[2]
+    if unit_vector_in_relations(i, M)
+      push!(to_delete, i)
+    end
+  end
+
+  new_generators = delete_columns(M_generators, to_delete)
+  new_relations = delete_columns(M_relations, to_delete)
+
+  to_delete,_,_ = rows_to_delete(vcat(new_generators, new_relations)',size(new_relations)[2])
+
+  new_generators = delete_columns(new_generators, to_delete)
+  new_relations = delete_columns(new_relations, to_delete)
+
+  #remove rows
+  #simplify relations
+  to_delete,_,_ = rows_to_delete(new_relations, size(new_relations)[1])
+
+  new_relations = delete_rows(new_relations, to_delete)
+
+  #simplify generators
+  to_delete, corresponding_row, K_gen = rows_to_delete(vcat(new_generators, new_relations), size(new_generators)[1])
+
+
+  injection_matrix = delete_rows(identity_matrix(R, size(M_generators)[1]), to_delete)
+  projection_matrix = zero_matrix(R, size(M_generators)[1], size(K_gen)[2]-length(to_delete))
+  for i=1:size(M_generators)[1]
+    if i in to_delete
+      index = findfirst(x -> x==i, to_delete)
+      assign_row!(projection_matrix, R(-1)*R(inv(coeff(K_gen[corresponding_row[index],i], 1)))*delete_columns(K_gen[corresponding_row[index],:], to_delete), i)
+    else
+      unit_vector_index = i-length(filter(x -> x < i, to_delete))
+      unit_vector = [j == unit_vector_index ? R(1) : R(0) for j=1:size(projection_matrix)[2]]
+      assign_row!(projection_matrix, unit_vector, i)
+    end
+  end
+
+  new_generators = delete_rows(new_generators, to_delete)
+
+  if length(new_generators)==0
+    zero_module = FreeMod(R,0)
+    injection = FreeModuleHom(zero_module, M, [])
+    projection = SubQuoHom(M, zero_module, [zero(zero_module) for i=1:ngens(M)])
+    # TODO early return or register morphisms?
+    return zero_module,injection,projection
+  else
+    SQ = iszero(new_relations) ? SubQuo(SubModuleOfFreeModule(new_generators)) : SubQuo(new_generators, new_relations)
+    injection = SubQuoHom(SQ, M, injection_matrix)
+    projection = SubQuoHom(M, SQ, projection_matrix[:,1:size(projection_matrix)[2]-size(new_relations)[1]])
+  end
+  register_morphism!(injection)
+  register_morphism!(projection)
+  injection.inverse_isomorphism = projection
+  projection.inverse_isomorphism = injection
+
+  #=if isgraded(M)
+    ishomogeneous_function_lookup_table[SQ] = x -> custom_ishomogeneous(injection(x))
+    degree_function_lookup_table[SQ] = x -> custom_degree(injection(x))
+  end=#
+
+  return SQ,injection,projection
+end
+
+######################################
+# Not only for testing
+######################################
+@doc Markdown.doc"""
+  matrix_to_map(F::FreeMod{T}, A::MatElem{T}) where T
+Converts a given n×m-matrix into the corresponding morphism $A : R^n → F$, 
```

Comment:
Let's not mix unicode and LaTeX:
```suggestion
Converts a given n×m-matrix into the corresponding morphism $A : R^n \to F$, 
```
Please change this everywhere.